### PR TITLE
Add extern "C" to isr_usbctrl in dev_lowlevel

### PR DESCRIPTION
**Pull request by supsm**
The USB dev_lowlevel example silently fails in C++ because `isr_usbctrl` is name-mangled and therefore not registered as an interrupt handler. This PR adds `extern "C"` to use C linkage for this function and allow compiling as C++.  

It is certainly not the prettiest, but it is necessary for correct behavior (with C++ linkage, none of the host's packets are responded to and the device is useless). The alternative to this is to manually specify `irq_set_exclusive_handler`, probably within `usb_device_init`. The pico-sdk advertises itself as C/C++, so C++ compatibility should be taken into consideration in its examples.

### CODE REVIEW

**Review by peterharperuk — APPROVED** [review]

File: usb/device/dev_lowlevel/dev_lowlevel.c [hunk 1]
```diff
@@ -489,6 +489,9 @@ static void usb_handle_buff_status() {
  *
  */
 /// \tag::isr_setup_packet[]
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (lurch) [inline usb/device/dev_lowlevel/dev_lowlevel.c:494]: Should this snippet be moved **above** the `/// \tag::isr_setup_packet[]` line? I don't think we want this `#ifdef` stuff appearing in section 4.1.3.2.3. of https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf ?

**Review by peterharperuk — CHANGES_REQUESTED** [review]
See comment from @lurch

**Comment by kilograham** [comment]
I'm not sure I understand why/how are you trying to compile this example as C++.(or what you mean by it).

File: usb/device/dev_lowlevel/dev_lowlevel.c (hunk 1 above)
Comment (supsm) [reply usb/device/dev_lowlevel/dev_lowlevel.c:493]: I wasn't aware of what that did. It should be fixed now.

Comment (peterharperuk) [reply usb/device/dev_lowlevel/dev_lowlevel.c:493]: Me neither!

**Comment by supsm** [comment]
> I'm not sure I understand why/how are you trying to compile this example as C++

I initially saved the file as .cpp to modify it in a more familiar language (not realizing isr_usbctrl is a special function). I suspect other developers may also attempt to unsuccessfully incorporate isr_usbctrl into C++ code.

**Comment by supsm** [comment]
Is there any update on this pr? I have already made the requested changes.

**Review by peterharperuk — APPROVED** [review]

**Pull request merged**